ansible_navigator/configuration/configuration.py

```diff
@@ -0,0 +1,143 @@
+""" The configuration object
+"""
+import os
+
+from copy import deepcopy
+from typing import List
+
+from .definitions import EntrySource
+from .definitions import Message
+from .parser import Parser
+
+from ..utils import error_and_exit_early
+from ..yaml import SafeLoader
+from ..yaml import yaml
+
+
+class Configuration:
+    # pylint: disable=too-many-arguments
+    # pylint: disable=too-few-public-methods
+    """the configuration class"""
+
+    def __init__(
+        self,
+        params,
+        application_configuration,
+        apply_previous_cli=False,
+        save_as_intitial=False,
+        settings_file_path=None,
+    ):
```

Comment:
Doc string for parameters will be helpful here :-)

Reply:
of course